src/codeFlowLogin.ts

```diff
@@ -231,7 +232,10 @@ function createTerminateServer(server: http.Server) {
 		});
 	});
 	return async () => {
-		const result = new Promise<void>(resolve => server.close(resolve));
+		const result = new Promise<void>((resolve, reject) => {
+			server.close(reject)
+			resolve();
```

Comment:
By calling `resolve` here, that means the `new Promise` will resolve immediately instead of waiting until the `server.close` event happens, which could again cause timing issues.

This is what we normally do with an `err?: Error | undefined` callback:
```typescript
const result = new Promise<void>((resolve, reject) => server.close(error => {
	if (error) {
		reject(error);
	} else {
		resolve();
	}
}));
```
_However_, that piece of code would throw an error whereas the original code would ignore the error. It's entirely possible ignoring the error was intentional behavior and that's why I do _not_ want to do this during an engineering PR. Most of this PR is sorting imports and messing with the types, which can't affect functionality. We can mess with the types to fix this line, too:

```typescript
const result = new Promise<void>((resolve: () => void) => server.close(resolve));
```